core/src/main/java/io/questdb/cairo/TableUtils.java

```diff
@@ -225,16 +290,31 @@ public static int exists(FilesFacade ff, Path path, CharSequence root, CharSeque
         }
     }
 
+    public static void freeTransitionIndex(long address) {
+        if (address == 0) {
+            return;
+        }
+        Unsafe.free(address, Unsafe.getUnsafe().getInt(address), MemoryTag.NATIVE_DEFAULT);
+    }
+
+    public static long getColumnHash(MemoryR metaMem, int columnIndex) {
+        return metaMem.getLong(META_OFFSET_COLUMN_TYPES + columnIndex * META_COLUMN_DATA_SIZE + 16);
+    }
+
     public static long getColumnNameOffset(int columnCount) {
         return META_OFFSET_COLUMN_TYPES + columnCount * META_COLUMN_DATA_SIZE;
     }
 
-    public static int getColumnType(ReadOnlyColumn metaMem, int columnIndex) {
-        return metaMem.getByte(META_OFFSET_COLUMN_TYPES + columnIndex * META_COLUMN_DATA_SIZE);
+    public static int getColumnType(MemoryR metaMem, int columnIndex) {
+        return metaMem.getInt(META_OFFSET_COLUMN_TYPES + columnIndex * META_COLUMN_DATA_SIZE);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).